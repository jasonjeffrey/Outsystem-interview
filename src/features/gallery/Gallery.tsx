import React, {useEffect} from 'react';
import {Grid,} from '@material-ui/core';
import {ImageGrid} from './components/image-grid/ImageGrid';
import {TreeView} from './components/tree-view/TreeView';
import {fetchFolders, GallerySliceState} from './gallerySlice';
import {useDispatch, useSelector} from 'react-redux';
import {ImageDataTable} from './components/image-data-table/ImageDataTable';
import styled from 'styled-components';

const TreeViewWrapper = styled.div`
  height: 50vh;
  overflow: scroll;
`;

const ImageDataTableWrapper = styled.div`
  height: 50vh;
`;

function Gallery() {
    const dispatch = useDispatch();
    const folders = useSelector((state: {gallery: GallerySliceState}) => state.gallery.folders);
    const artist = useSelector((state: {gallery: GallerySliceState}) => state.gallery.selectedArtist);
    const selectedImage = useSelector((state: {gallery: GallerySliceState}) => state.gallery.selectedImage);

    useEffect(() => {
        dispatch(fetchFolders());
    }, [dispatch]);

    return (
        <Grid container style={{height: '100vh'}}>
            <Grid item xs={9}>
                <h1>{artist?.name}</h1>
                <ImageGrid images={artist?.portfolio} selectedIndex={selectedImage?.id}/>
            </Grid>
            <Grid item xs={3}>
                <TreeViewWrapper>
                    <TreeView folders={folders} />
                </TreeViewWrapper>
                <ImageDataTableWrapper>
                    <ImageDataTable
                        url={selectedImage?.url}
                        width={selectedImage?.width}
                        height={selectedImage?.height}
                        xResolution={selectedImage?.xResolution}
                        yResolution={selectedImage?.yResolution}/>
                </ImageDataTableWrapper>

            </Grid>
        </Grid>
    );
}

export default Gallery;
