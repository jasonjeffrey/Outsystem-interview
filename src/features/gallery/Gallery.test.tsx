import React from 'react';
import { render} from '@testing-library/react';
import Gallery from './Gallery';

jest.mock('react-redux', () => ({
    useSelector: () => [],
    useDispatch: () => jest.fn()
}));

describe("Gallery Feature", () => {
    it('RENDER Gallery Component', () => {
        const component = <Gallery/>
        const {asFragment} = render(component);

        expect(asFragment()).toMatchSnapshot();
    });
})
