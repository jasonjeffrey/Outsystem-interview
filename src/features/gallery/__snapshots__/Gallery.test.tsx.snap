// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Feature RENDER Gallery Component 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
    style="height: 100vh;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
    >
      <h1 />
      <ul
        class="MuiGridList-root"
        style="margin: -2px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <ul
        class="MuiList-root MuiList-padding"
      />
      <div
        class="MuiTableContainer-root"
      >
        <table
          class="MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Url
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Width
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Height
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                XResolution
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                YResolution
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
